PlatformControl implementation for Adwaita theme

fujii opened the pull request:
#### 15c1edc2edeafd19a0abfc50188402128a916afa
<pre>
PlatformControl implementation for Adwaita theme
<a href="https://bugs.webkit.org/show_bug.cgi?id=278154">https://bugs.webkit.org/show_bug.cgi?id=278154</a>

Reviewed by Carlos Garcia Campos.

Moved painting code from RenderThemeAdwaita to PlatformControl
subclasses. It&apos;s more efficient in GPU process mode.

* Source/WebCore/platform/Adwaita.cmake:
* Source/WebCore/platform/SourcesAdwaita.txt:
* Source/WebCore/platform/Theme.h:
(WebCore::Theme::paint): Deleted.
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
(): Deleted.
(WebCore::ThemeAdwaita::focusColor): Deleted.
(WebCore::getRectRadius): Deleted.
(WebCore::ThemeAdwaita::paintFocus): Deleted.
(WebCore::ThemeAdwaita::paintArrow): Deleted.
(WebCore::ThemeAdwaita::paint): Deleted.
(WebCore::ThemeAdwaita::paintCheckbox): Deleted.
(WebCore::ThemeAdwaita::paintRadio): Deleted.
(WebCore::ThemeAdwaita::paintButton): Deleted.
(WebCore::ThemeAdwaita::paintSpinButton): Deleted.
* Source/WebCore/platform/adwaita/ThemeAdwaita.h:
* Source/WebCore/platform/graphics/adwaita/Adwaita.cpp: Added.
(WebCore::Adwaita::getRectRadius):
(WebCore::Adwaita::paintFocus):
(WebCore::Adwaita::paintArrow):
(WebCore::Adwaita::systemAccentColor):
(WebCore::Adwaita::systemFocusRingColor):
* Source/WebCore/platform/graphics/adwaita/Adwaita.h: Added.
(WebCore::Adwaita::focusColor):
* Source/WebCore/platform/graphics/adwaita/ButtonControlAdwaita.cpp: Added.
(WebCore::ButtonControlAdwaita::ButtonControlAdwaita):
(WebCore::ButtonControlAdwaita::draw):
(WebCore::ButtonControlAdwaita::drawButton):
* Source/WebCore/platform/graphics/adwaita/ButtonControlAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ControlAdwaita.cpp: Added.
(WebCore::ControlAdwaita::ControlAdwaita):
(WebCore::ControlAdwaita::accentColor):
* Source/WebCore/platform/graphics/adwaita/ControlAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ControlFactoryAdwaita.cpp: Added.
(WebCore::ControlFactory::create):
(WebCore::ControlFactoryAdwaita::shared):
(WebCore::ControlFactoryAdwaita::createPlatformApplePayButton):
(WebCore::ControlFactoryAdwaita::createPlatformButton):
(WebCore::ControlFactoryAdwaita::createPlatformColorWell):
(WebCore::ControlFactoryAdwaita::createPlatformImageControlsButton):
(WebCore::ControlFactoryAdwaita::createPlatformInnerSpinButton):
(WebCore::ControlFactoryAdwaita::createPlatformMenuList):
(WebCore::ControlFactoryAdwaita::createPlatformMenuListButton):
(WebCore::ControlFactoryAdwaita::createPlatformMeter):
(WebCore::ControlFactoryAdwaita::createPlatformProgressBar):
(WebCore::ControlFactoryAdwaita::createPlatformSearchField):
(WebCore::ControlFactoryAdwaita::createPlatformSearchFieldCancelButton):
(WebCore::ControlFactoryAdwaita::createPlatformSearchFieldResults):
(WebCore::ControlFactoryAdwaita::createPlatformSliderThumb):
(WebCore::ControlFactoryAdwaita::createPlatformSliderTrack):
(WebCore::ControlFactoryAdwaita::createPlatformSwitchThumb):
(WebCore::ControlFactoryAdwaita::createPlatformSwitchTrack):
(WebCore::ControlFactoryAdwaita::createPlatformTextArea):
(WebCore::ControlFactoryAdwaita::createPlatformTextField):
(WebCore::ControlFactoryAdwaita::createPlatformToggleButton):
* Source/WebCore/platform/graphics/adwaita/ControlFactoryAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/InnerSpinButtonAdwaita.cpp: Added.
(WebCore::InnerSpinButtonAdwaita::InnerSpinButtonAdwaita):
(WebCore::InnerSpinButtonAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/InnerSpinButtonAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/MenuListAdwaita.cpp: Added.
(WebCore::MenuListAdwaita::MenuListAdwaita):
(WebCore::MenuListAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/MenuListAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ProgressBarAdwaita.cpp: Added.
(WebCore::ProgressBarAdwaita::ProgressBarAdwaita):
(WebCore::currentAnimationProgress):
(WebCore::ProgressBarAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/ProgressBarAdwaita.h: Added.
(WebCore::ProgressBarAdwaita::owningProgressBarPart const):
* Source/WebCore/platform/graphics/adwaita/SliderThumbAdwaita.cpp: Added.
(WebCore::SliderThumbAdwaita::SliderThumbAdwaita):
(WebCore::SliderThumbAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/SliderThumbAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/SliderTrackAdwaita.cpp: Added.
(WebCore::SliderTrackAdwaita::SliderTrackAdwaita):
(WebCore::SliderTrackAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/SliderTrackAdwaita.h: Added.
(WebCore::SliderTrackAdwaita::owningSliderTrackPart const):
* Source/WebCore/platform/graphics/adwaita/TextFieldAdwaita.cpp: Added.
(WebCore::TextFieldAdwaita::TextFieldAdwaita):
(WebCore::TextFieldAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/TextFieldAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ToggleButtonAdwaita.cpp: Added.
(WebCore::ToggleButtonAdwaita::ToggleButtonAdwaita):
(WebCore::ToggleButtonAdwaita::draw):
(WebCore::ToggleButtonAdwaita::drawCheckbox):
(WebCore::ToggleButtonAdwaita::drawRadio):
* Source/WebCore/platform/graphics/adwaita/ToggleButtonAdwaita.h: Added.
* Source/WebCore/platform/graphics/cairo/GraphicsContextCairo.cpp:
(WebCore::GraphicsContextCairo::drawFocusRing):
* Source/WebCore/platform/graphics/controls/ControlFactory.cpp:
* Source/WebCore/platform/graphics/nicosia/cairo/NicosiaCairoOperationRecorder.cpp:
(Nicosia::CairoOperationRecorder::drawFocusRing):
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
(WebCore::GraphicsContextSkia::drawFocusRing):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paint):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::getAccentColor):
(WebCore::RenderThemeAdwaita::canCreateControlPartForRenderer const):
(WebCore::RenderThemeAdwaita::canCreateControlPartForBorderOnly const):
(WebCore::RenderThemeAdwaita::canCreateControlPartForDecorations const):
(WebCore::RenderThemeAdwaita::platformActiveSelectionBackgroundColor const):
(WebCore::RenderThemeAdwaita::platformFocusRingColor const):
(): Deleted.
(WebCore::getSystemAccentColor): Deleted.
(WebCore::RenderThemeAdwaita::paintTextField): Deleted.
(WebCore::RenderThemeAdwaita::paintTextArea): Deleted.
(WebCore::RenderThemeAdwaita::paintSearchField): Deleted.
(WebCore::RenderThemeAdwaita::paintMenuList): Deleted.
(WebCore::RenderThemeAdwaita::paintMenuListButtonDecorations): Deleted.
(WebCore::RenderThemeAdwaita::animationDurationForProgressBar const): Deleted.
(WebCore::RenderThemeAdwaita::paintProgressBar): Deleted.
(WebCore::RenderThemeAdwaita::paintSliderTrack): Deleted.
(WebCore::RenderThemeAdwaita::paintSliderThumb): Deleted.
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:

Canonical link: <a href="https://commits.webkit.org/282487@main">https://commits.webkit.org/282487@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2c25f452dfb305881fb4f83736418fad8f82cd80

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63299 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/42655 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/15896 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67320 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/13907 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/65419 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/50343 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14187 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/50990 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9607 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66368 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/39603 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/54815 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/31673 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/36286 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12164 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/12779 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/57825 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12492 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69016 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7246 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12095 "Found 1 new test failure: media/modern-media-controls/tracks-support/no-tracks.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58302 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7277 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/54885 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58531 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6037 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/9561 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38476 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/39555 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/40667 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39298 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebCore/platform/graphics/adwaita/Adwaita.h:
```diff
@@ -0,0 +1,137 @@
+constexpr auto spinButtonBackgroundPressedColorDark = SRGBA<uint8_t> { 255, 255, 255, 70 };
+
+
+enum class PaintRounded : bool { No, Yes };
+void paintFocus(GraphicsContext&, const FloatRect&, int offset, const Color&, PaintRounded = PaintRounded::No);
+void paintFocus(GraphicsContext&, const Path&, const Color&);
+void paintFocus(GraphicsContext&, const Vector<FloatRect>&, const Color&, PaintRounded = PaintRounded::No);
+
+enum class ArrowDirection { Up, Down };
+void paintArrow(GraphicsContext&, const FloatRect&, ArrowDirection, bool useDarkAppearance);
+
+inline Color focusColor(const Color& accentColor)
+{
+    return accentColor.colorWithAlphaMultipliedBy(focusRingOpacity);
+}
+
+Color systemAccentColor();
+Color systemFocusRingColor();
```
I guess these global methods could be part of the base ControlAdwaita, no?

fujii replied:
Yes, it could. But, `paintFocus` is called from some classes `GraphicsContextCairo`, `GraphicsContextSkia` and `NicosiaCairoOperationRecorder`.
These classes have to include `ControlAdwaita.h` and call `ControlAdwaita::paintFocus()`.
I though including `Adwaita.h` and calling `Adwaita::paintFocus()` are a bit more concise. But, not a strong opinion. I will change if you prefer.

carlosgcampos commented on Source/WebCore/platform/graphics/adwaita/ControlAdwaita.h:
```diff
@@ -0,0 +1,52 @@
+    ControlAdwaita(ControlPart&, ControlFactoryAdwaita&);
+
+protected:
+    static Color getAccentColor(const ControlStyle&);
```
```suggestion
    static Color accentColor(const ControlStyle&);
```

carlosgcampos replied:
Why is this one here but systemAccentColor and others are global in an Adwaita namespace?

fujii replied:
`Adwaita.h` is included from some places.
Only this function depends on `ControlStyle` class. I just reduced the number of headers to be included recursively.
But, not important, nor a strong opinion. Which do you prefer, ControlAdwaita class static functions or functions in Adwaita namespace?
I will rename getAccentColor.

carlosgcampos replied:
It's ok, if it reduces the header includes.

carlosgcampos commented on Source/WebCore/platform/graphics/adwaita/ControlAdwaita.h:
```diff
@@ -0,0 +1,52 @@
+namespace WebCore {
+
+class ControlFactoryAdwaita;
+class Path;
```
I don't see Path used in this header

carlosgcampos commented on Source/WebCore/platform/graphics/adwaita/ButtonControlAdwaita.h:
```diff
@@ -0,0 +1,45 @@
+
+namespace WebCore {
+
+class ButtonControlAdwaita : public ControlAdwaita {
```
```suggestion
class ButtonControlAdwaita final : public ControlAdwaita {
```

carlosgcampos replied:
And the same for all other classes that are final

fujii reviewed (commented):
Thank you very much for the review.

fujii committed:
PlatformControl implementation for Adwaita theme
https://bugs.webkit.org/show_bug.cgi?id=278154

Reviewed by Carlos Garcia Campos.

Moved painting code from RenderThemeAdwaita to PlatformControl
subclasses. It's more efficient in GPU process mode.

* Source/WebCore/platform/Adwaita.cmake:
* Source/WebCore/platform/SourcesAdwaita.txt:
* Source/WebCore/platform/Theme.h:
(WebCore::Theme::paint): Deleted.
* Source/WebCore/platform/adwaita/ThemeAdwaita.cpp:
(): Deleted.
(WebCore::ThemeAdwaita::focusColor): Deleted.
(WebCore::getRectRadius): Deleted.
(WebCore::ThemeAdwaita::paintFocus): Deleted.
(WebCore::ThemeAdwaita::paintArrow): Deleted.
(WebCore::ThemeAdwaita::paint): Deleted.
(WebCore::ThemeAdwaita::paintCheckbox): Deleted.
(WebCore::ThemeAdwaita::paintRadio): Deleted.
(WebCore::ThemeAdwaita::paintButton): Deleted.
(WebCore::ThemeAdwaita::paintSpinButton): Deleted.
* Source/WebCore/platform/adwaita/ThemeAdwaita.h:
* Source/WebCore/platform/graphics/adwaita/Adwaita.cpp: Added.
(WebCore::Adwaita::getRectRadius):
(WebCore::Adwaita::paintFocus):
(WebCore::Adwaita::paintArrow):
(WebCore::Adwaita::systemAccentColor):
(WebCore::Adwaita::systemFocusRingColor):
* Source/WebCore/platform/graphics/adwaita/Adwaita.h: Added.
(WebCore::Adwaita::focusColor):
* Source/WebCore/platform/graphics/adwaita/ButtonControlAdwaita.cpp: Added.
(WebCore::ButtonControlAdwaita::ButtonControlAdwaita):
(WebCore::ButtonControlAdwaita::draw):
(WebCore::ButtonControlAdwaita::drawButton):
* Source/WebCore/platform/graphics/adwaita/ButtonControlAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ControlAdwaita.cpp: Added.
(WebCore::ControlAdwaita::ControlAdwaita):
(WebCore::ControlAdwaita::accentColor):
* Source/WebCore/platform/graphics/adwaita/ControlAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ControlFactoryAdwaita.cpp: Added.
(WebCore::ControlFactory::create):
(WebCore::ControlFactoryAdwaita::shared):
(WebCore::ControlFactoryAdwaita::createPlatformApplePayButton):
(WebCore::ControlFactoryAdwaita::createPlatformButton):
(WebCore::ControlFactoryAdwaita::createPlatformColorWell):
(WebCore::ControlFactoryAdwaita::createPlatformImageControlsButton):
(WebCore::ControlFactoryAdwaita::createPlatformInnerSpinButton):
(WebCore::ControlFactoryAdwaita::createPlatformMenuList):
(WebCore::ControlFactoryAdwaita::createPlatformMenuListButton):
(WebCore::ControlFactoryAdwaita::createPlatformMeter):
(WebCore::ControlFactoryAdwaita::createPlatformProgressBar):
(WebCore::ControlFactoryAdwaita::createPlatformSearchField):
(WebCore::ControlFactoryAdwaita::createPlatformSearchFieldCancelButton):
(WebCore::ControlFactoryAdwaita::createPlatformSearchFieldResults):
(WebCore::ControlFactoryAdwaita::createPlatformSliderThumb):
(WebCore::ControlFactoryAdwaita::createPlatformSliderTrack):
(WebCore::ControlFactoryAdwaita::createPlatformSwitchThumb):
(WebCore::ControlFactoryAdwaita::createPlatformSwitchTrack):
(WebCore::ControlFactoryAdwaita::createPlatformTextArea):
(WebCore::ControlFactoryAdwaita::createPlatformTextField):
(WebCore::ControlFactoryAdwaita::createPlatformToggleButton):
* Source/WebCore/platform/graphics/adwaita/ControlFactoryAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/InnerSpinButtonAdwaita.cpp: Added.
(WebCore::InnerSpinButtonAdwaita::InnerSpinButtonAdwaita):
(WebCore::InnerSpinButtonAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/InnerSpinButtonAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/MenuListAdwaita.cpp: Added.
(WebCore::MenuListAdwaita::MenuListAdwaita):
(WebCore::MenuListAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/MenuListAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ProgressBarAdwaita.cpp: Added.
(WebCore::ProgressBarAdwaita::ProgressBarAdwaita):
(WebCore::currentAnimationProgress):
(WebCore::ProgressBarAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/ProgressBarAdwaita.h: Added.
(WebCore::ProgressBarAdwaita::owningProgressBarPart const):
* Source/WebCore/platform/graphics/adwaita/SliderThumbAdwaita.cpp: Added.
(WebCore::SliderThumbAdwaita::SliderThumbAdwaita):
(WebCore::SliderThumbAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/SliderThumbAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/SliderTrackAdwaita.cpp: Added.
(WebCore::SliderTrackAdwaita::SliderTrackAdwaita):
(WebCore::SliderTrackAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/SliderTrackAdwaita.h: Added.
(WebCore::SliderTrackAdwaita::owningSliderTrackPart const):
* Source/WebCore/platform/graphics/adwaita/TextFieldAdwaita.cpp: Added.
(WebCore::TextFieldAdwaita::TextFieldAdwaita):
(WebCore::TextFieldAdwaita::draw):
* Source/WebCore/platform/graphics/adwaita/TextFieldAdwaita.h: Added.
* Source/WebCore/platform/graphics/adwaita/ToggleButtonAdwaita.cpp: Added.
(WebCore::ToggleButtonAdwaita::ToggleButtonAdwaita):
(WebCore::ToggleButtonAdwaita::draw):
(WebCore::ToggleButtonAdwaita::drawCheckbox):
(WebCore::ToggleButtonAdwaita::drawRadio):
* Source/WebCore/platform/graphics/adwaita/ToggleButtonAdwaita.h: Added.
* Source/WebCore/platform/graphics/cairo/GraphicsContextCairo.cpp:
(WebCore::GraphicsContextCairo::drawFocusRing):
* Source/WebCore/platform/graphics/controls/ControlFactory.cpp:
* Source/WebCore/platform/graphics/nicosia/cairo/NicosiaCairoOperationRecorder.cpp:
(Nicosia::CairoOperationRecorder::drawFocusRing):
* Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp:
(WebCore::GraphicsContextSkia::drawFocusRing):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::paint):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::getAccentColor):
(WebCore::RenderThemeAdwaita::canCreateControlPartForRenderer const):
(WebCore::RenderThemeAdwaita::canCreateControlPartForBorderOnly const):
(WebCore::RenderThemeAdwaita::canCreateControlPartForDecorations const):
(WebCore::RenderThemeAdwaita::platformActiveSelectionBackgroundColor const):
(WebCore::RenderThemeAdwaita::platformFocusRingColor const):
(): Deleted.
(WebCore::getSystemAccentColor): Deleted.
(WebCore::RenderThemeAdwaita::paintTextField): Deleted.
(WebCore::RenderThemeAdwaita::paintTextArea): Deleted.
(WebCore::RenderThemeAdwaita::paintSearchField): Deleted.
(WebCore::RenderThemeAdwaita::paintMenuList): Deleted.
(WebCore::RenderThemeAdwaita::paintMenuListButtonDecorations): Deleted.
(WebCore::RenderThemeAdwaita::animationDurationForProgressBar const): Deleted.
(WebCore::RenderThemeAdwaita::paintProgressBar): Deleted.
(WebCore::RenderThemeAdwaita::paintSliderTrack): Deleted.
(WebCore::RenderThemeAdwaita::paintSliderThumb): Deleted.
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:

Canonical link: https://commits.webkit.org/282487@main
